### PR TITLE
Vebt 40 applicant info

### PR DESCRIPTION
**Pull request by wafimohamed**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
- created all fields for form for 22-10282. also created the Confirmation page as well as Review/Submission  page
- _(If bug, how to reproduce)_
- _(What is the solution, why is this the solution)_
- _(Which team do you work for, does your team own the maintenance of this component?)_
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

### **VEBT-40**
As a...

VEBT Business User

I want…

to fill out the form 22-10282 online with my applicant info

So that…

I can submit the applicant information 

Acceptance Criteria

NOTE:- This is broken down by Sections within the STEP. Each SECTION is its own page and the user will move onto the next once all REQ fields are inputted. 

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 1

H3 -> *Your name*

Your first name *Required
Your middle initial
Your last name *Required

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 2

H3 -> Which of these best describes you? *Required

I'm a Veteran
I'm a Veteran's Spouse
I'm a Veteran's Child
I'm a Veteran's Caregiver (who isn't their spouse or child)
I'm an active duty service member
I'm a member of the National Guard
I'm a Reservist
I'm a member of the Individual ready reserve

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 3

H3 -> Phone and Email Address

Email address *Required
Mobile phone number
Home phone number

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 4

H3 -> What country do you live in? ( This will be conditional -> skip state question to the next one if country is not USA)

Country *Required

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 5

H3 -> What state do you live in? ( If country is not USA, skip this question)

State *Required

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 7

The next few questions are about race, gender, education, and employment. These questions are optional. You don't need to answer them. We ask these questions for statistical purposes. Your answers won't affect your eligibility for the IBM SkillsBuild program.
Do you want to answer the optional questions? 

Yes
No

Note(s) from design team:

If 'Yes' is selected, the end-user will be directed to the race/gender questions. If 'No' is selected, proceed to step 2.

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 8

H3 -> Your ethnicity, race, or origin

What is your ethnicity?

Hispanic or Latino
Not Hispanic or Latino
Prefer not to answer

What is your race or origin?

select all that you identify with

American Indian or Alaskan Native
Asian
Black or African American
Native Hawaiian or Other Pacific Islander
White
Middle Eastern or North African
Other
Prefer not to answer

STEP 1 of 3: Your personal information ( NOTE:- this is only for Dev info and not to be added ) SECTION 9

H3 -> How would you describe your gender?

You can change your selection at any time. If you
decide you no longer want to share your gender
identity, select Prefer not to answer.

Man
Non-binary
Transgender man
Transgender woman
Woman
Prefer not to answer
A gender not listed here

Validation on STEP 1

Your first name  - Please enter your first name
Your last name  - Please enter your last name
 Sex of applicant (How would you describe your gender?)  - Please enter your gender 
Email address - Please enter your email address
Re-enter email address - Please enter your email address

The following field edits are displayed for invalid entries:

Email address -  Enter a valid email address using the format email@domain.com. Your email address can only have letters, numbers, the @ symbol and a period, with no spaces.
Re-enter email address -
Enter a valid email address using the format email@domain.com. Your email address can only have letters, numbers, the @ symbol and a period, with no spaces.
Please ensure your entries match
Mobile phone number - Please enter a valid phone number
Home phone number -  Please enter a valid phone number

Technical Details:

Dependencies: No

Feature Flag needed: Yes

Demo needed: Yes

UAT needed: Yes

Additional Notes: No conditional questions or sections. If the user Answers YES, NO, or leaves it blank the user will be shown the next page! 

Testing Details:

(  ) Quick Turn Around/Direct to Production

( X ) Remains in Staging

(  ) Staging Environment unavailable, coordinate w/ developer

Requested by and when:

Definition of Done: 

(  )  UAT completed with EDU

(  )  Passed all internal testing in Staging

(  )  Passed all internal testing in Production

(  )  All work documented in Jira ticket

### **VEBT-42**
As a...

VEBT Business User

I want…

to fill out the form 22-10282 online with my chosen program

So that…

I can submit the program information

Acceptance Criteria

Each section is its own page even though it may be on the same STEP of the application. 

STEP 2 of 3: Your education and employment history ( NOTE:- this is only for Dev info and not to be added ) SECTION 1

H3 -> What's the highest level of education you have completed?

A high school diploma or GED
An associate degree
A bachelor’s degree
A master’s degree
A doctoral degree like a PhD
Something else

If user selects Something else, a new text box will appear letting the user input info.

Notes from design team:
Note that this conditional reveal for "Something else" will not be required for user input for the first question on step 2/3
None of the questions in step 2 are required, so even if the end-user does not make a selection, the next page should still be displayed as they click Continue

STEP 2 of 3: Your education and employment history ( NOTE:- this is only for Dev info and not to be added ) SECTION 2

H3 -> Are you currently employed?

Yes
No 

STEP 2 of 3: Your education and employment history ( NOTE:- this is only for Dev info and not to be added ) SECTION 3

H3 -> What's your current annual salary?

Less than $20,000
Between $20,001 and $35,000
Between $35,001 and $50,000
Between $50,001 and $75,000
More than $75,000

STEP 2 of 3: Your education and employment history ( NOTE:- this is only for Dev info and not to be added ) SECTION 4

H3 -> Do you currently work in the technology industry?

Yes
No

H3 -> What's your main area of focus in the technology industry?

Computer programming
Data processing
Computer software
Information sciences
Media application
Something else not listed here

Technical Details:

Dependencies: No

Feature Flag needed: Yes

Demo needed: Yes

UAT needed: Yes

Additional Notes: No conditional questions or sections. If the user Answers YES, NO, or leaves it blank the user will be shown the next page! 

Testing Details:

(  ) Quick Turn Around/Direct to Production

( X ) Remains in Staging

(  ) Staging Environment unavailable, coordinate w/ developer

Requested by and when:

Definition of Done: 

(  )  UAT completed with EDU

(  )  Passed all internal testing in Staging

(  )  Passed all internal testing in Production

(  )  All work documented in Jira ticket

### **VEBT-29**

As a...

VEBT Business User

I want…

to fill out the form 22-10282 online  

So that…

I can submit the information 

Acceptance Criteria

STEP 3 of 3: Review ( NOTE:- this is only for Dev info and not to be added ) SECTION 1

User can see the Expand all sections (In this case, STEP 1 and STEP 2) are available to the user to expand. Default is collapsed. The user is able to review their entries and edit.
Below the Expand/review section of the form is the Statement of truth where the user will sign off on the application form digitally. 

H3 -> Statement of truth

I confirm that the identifying information in this form is accurate and has been represented correctly.
I have read and accept the privacy policy .
Your full name (*Required)

I certify that the information above is correct and true to the best of my knowledge and belief. (*Required)

Technical Details:

Dependencies: No

Feature Flag needed: No

Demo needed: No

UAT needed: No

Additional Notes:  

Testing Details:

( ) Quick Turn Around/Direct to Production

(  X ) Remains in Staging

(  ) Staging Environment unavailable, coordinate w/ developer

Requested by and when:

Definition of Done: 

(  )  Demo/UAT completed with EDU

(  )  Passed all internal testing in Staging

(  )  Passed all internal testing in Production

(  )  All work documented in Jira ticket

### **VEBT-23**

As a...

VEBT Business User

I want…

to view the confirmation of the 22-10282 submission

So that…

I have a record of my submission

Acceptance Criteria

After the applicant successfully submits the 22-10282,  the system displays a confirmation page which contains the ability to print the confirmation and a PDF of the submitted data.  The following information is displayed on the confirmation page:

We've received your application

We may contact you if we need more information.

IBM SkillsBuild Training Program Application (Form 22-10282)

For name of applicant

Confirmation number

 randomly generated number

Date received

Month day, year of submission

Your application was sent to vettectpartner@va.gov

What happens after I apply?

You'll receive a response soon with more information by email.

[Form 22-10282.xlsx (sharepoint.com)](https://dvagov.sharepoint.com/:x:/r/sites/VREPSES/_layouts/15/Doc.aspx?sourcedoc=%7B1097F495-329A-4DE7-9992-AFD64AEBA219%7D&file=Form%2022-10282.xlsx&action=default&mobileredirect=true&wdsle=0)

Technical Details:

Dependencies: No

Feature Flag needed: Yes

Demo needed: Yes

UAT needed: Yes

Additional Notes: 

Testing Details:

(  ) Quick Turn Around/Direct to Production

( X ) Remains in Staging

(  ) Staging Environment unavailable, coordinate w/ developer

Requested by and when:

Definition of Done: 

(  )  UAT completed with EDU

(  )  Passed all internal testing in Staging

(  )  Passed all internal testing in Production

(  )  All work documented in Jira ticket

- _Link to ticket created in va.gov-team repo_
- https://jira.devops.va.gov/browse/VEBT-40
- https://jira.devops.va.gov/browse/VEBT-42
- https://jira.devops.va.gov/browse/VEBT-29
- https://jira.devops.va.gov/browse/VEBT-23
department-of-veterans-affairs/va.gov-team#0000
- _Link to previous change of the code/bug (if applicable)_
department-of-veterans-affairs/vets-website#0000
- _Link to epic if not included in ticket_
department-of-veterans-affairs/va.gov-team#0000

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution r
image
ecords](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

| ------- | ------ | ----- |
| Mobile  |        
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image

      |
| Desktop |      
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image

         |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by jsimonVA — COMMENTED** [review]
### Personal Information page
- The size of the Middle initial Field does not match the width in Figma. Please check with Ariana on the correct size
image

- Same with the back button, it seems too wide. Did you add width to this? This is throughout the form
image

- Individual Ready Reserve needs to be capitalized as shown in Mockup
image

- Double check capitalization. Email is not capitalized in the mockup but it is on here
image

- this is Different than what is in the Mockup. The Mock up has the question, "What country do you live in?" then the title of "Country" with the "required" text next to "Country" Double check this page against mockup. Same with the State question.
image

- Double check verbiage, it doesn't match what is in the Mockup
image

- Need spacing below "Your ethnicity, race, or orgin". See Mockup.
image

### Your education and employment history
- This is small, but in the mockup we have "you've" and on here we have "you have". I think we should match the mockup since it is whats been approved.
image

### Confirmation Page
- In the What to Expect Next section on the bottom, look at the mockup. There should be two separate paragraphs. Currently it in just one big one. Also, the email is incorrect. Make sure the email is `sbuser@us.ibm.com` and that it is coded as an email address.
image

File: src/applications/edu-benefits/10282/pages/applicantInformation/applicantDescription.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+
+const labels = {
+  veteran: "I'm a Veteran",
+  veteranSpouse: "I'm a Veteran's Spouse",
+  veteranChild: "I'm a Veteran's Child",
+  veteranCaregiver:
+    "I'm a Veteran's Caregiver (who isn't their spouse or child)",
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/pages/applicantInformation/applicantDescription.js:8]: `spouse`, `child`, and `caregiver` should all be lowercase here

File: src/applications/edu-benefits/10282/pages/applicantInformation/genderRaceQuestion.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import fullSchema10282 from 'vets-json-schema/dist/22-10282-schema.json';
+
+const { raceAndGender } = fullSchema10282.definitions;
+const uiSchema = {
+  'ui:description':
+    "The next few questions are about race, gender, education, and employment. These questions are optional. You don't need to answer them. We ask these questions for statistical purposes. Your answers won't affect your eligibility for the IBM SkillsBuild program.",
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/pages/applicantInformation/genderRaceQuestion.js:6]: `We ask these questions for ...` should be it's own paragraph based on the latest Figma design

File: src/applications/edu-benefits/10282/pages/applicantInformation/applicantRaceAndEthnicity.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  ),
+  ethnicity: {
+    'ui:title': 'What is your ethnicity?',
+    'ui:widget': 'radio',
+  },
+  orginRace: {
+    'ui:title': 'What is your race or origin?',
+    'ui:description': (
+      <p className="vads-u-margin-top--0  vads-u-color--gray-medium">
+        select all that you identify with
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/pages/applicantInformation/applicantRaceAndEthnicity.js:18]: `Select all ...` should be capitalized

File: src/applications/edu-benefits/10282/config/form.js [hunk 4]
```diff
@@ -1,144 +1,195 @@
...
+        },
+        currentlyEmployed: {
+          title: 'Your education and employment history',
+          path: 'applicant-information-9',
           uiSchema: {
-            fullName: fullNameUI,
-            ssn: ssnUI,
+            currentlyEmployed: {
+              'ui:title': (
+                <h3 className="vads-u-margin--0">Are you currently employed</h3>
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/config/form.js:143]: Should this have a question mark on it?

File: src/applications/edu-benefits/10282/config/form.js [hunk 5]
```diff
@@ -1,144 +1,195 @@
...
+          schema: applicantGender.schema,
+          depends: formData => formData.raceAndGender === 'Yes',
+        },
+      },
+    },
+    educationAndEmploymentHistory: {
+      title: ' Your education and employment history',
+      pages: {
+        highestLevelOfEducation: {
+          title: 'Your education and employment history',
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/config/form.js:132]: On mobile, the text should be styled to be left aligned only up to the step number.

image

File: src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx [hunk 6]
```diff
@@ -52,13 +52,42 @@ export class ConfirmationPage extends React.Component {
...
+            filetype="PDF"
+            href="/education/apply-for-education-benefits/10282/confirmation"
+            text="Download your completed application"
+          />
+        </div>
+        <div className="vads-u-margin-top--0">
+          <h2>What to expect next</h2>
+          <p>
+            If you're accepted into the SkillsBuild program, you'll receive an
+            email from souser@us.ibm.com with your login informatia. If you
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx:83]: Should the email be an achor tag with a mailto?

File: src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx [hunk 7]
```diff
@@ -52,13 +52,42 @@ export class ConfirmationPage extends React.Component {
...
+            don't receive an email, check your spam folder. You'll also receive
+            an email from SkillUp Online, SkillsBuild's training provider, with
+            more resources and support for your training. SkillUp Online will
+            help you choose a learning format for your training. You can choose
+            between independent learning (also known as self-paced learning), or
+            cohort learning (which includes scheduled online classes).
```
Comment (mghisilieri) [inline src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx:89]: A new paragraph should start at `You'll also receive ...`

**Review by mghisilieri — COMMENTED** [review]
Here are some general comments just from taking a look at the mobile screenshots. The only thing where I didn't find relevant code was naming the `Review` page as such:

image

File: src/applications/edu-benefits/10282/config/form.js (hunk 4 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/config/form.js:143]: I Think so but most of those did copy and baste from Jira lol

File: src/applications/edu-benefits/10282/config/form.js (hunk 5 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/config/form.js:132]: other forms have the same thing. it's not aligned

File: src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx (hunk 6 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx:83]: done

File: src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx (hunk 7 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/containers/ConfirmationPage.jsx:89]: done

File: src/applications/edu-benefits/10282/config/form.js (hunk 4 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/config/form.js:143]: done

File: src/applications/edu-benefits/10282/pages/applicantInformation/applicantRaceAndEthnicity.js (hunk 3 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/pages/applicantInformation/applicantRaceAndEthnicity.js:18]: done

File: src/applications/edu-benefits/10282/pages/applicantInformation/genderRaceQuestion.js (hunk 2 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/pages/applicantInformation/genderRaceQuestion.js:6]: done

File: src/applications/edu-benefits/10282/pages/applicantInformation/applicantDescription.js (hunk 1 above)
Comment (wafimohamed) [reply src/applications/edu-benefits/10282/pages/applicantInformation/applicantDescription.js:8]: done

**Review by GovCIOLiz — COMMENTED** [review]
In addition to comments let's also review the review section - we got repeated text that we don't have in the mockups and some font that's bigger than we need it throughout Screenshot 2024-09-27 at 9 59 17 AM
Screenshot 2024-09-27 at 9 59 43 AM

**Review by GovCIOLiz — DISMISSED** [review]

**Review by GovCIOLiz — APPROVED** [review]

**Review by pjhill — APPROVED** [review]

**Pull request merged**